### PR TITLE
[NCL-8655] Fix SCM link name and refactor urlParseHelper

### PR DESCRIPTION
**Pull request by DnsZhou**

### CODE REVIEW

File: src/utils/urlParseHelper.ts [hunk 1]
```diff
@@ -2,58 +2,48 @@ interface IScmRepositoryUrl {
...
+  const name: string = preDefinedScms[base] || base;
+
+  // Special handling for URLs from Gerrit, will be removed after Gerrit support ends.
+  const matchedGerritUrl = Object.keys(preDefinedScms).find((key) => key.includes(base) && preDefinedScms[key] === 'Gerrit');
+  if (matchedGerritUrl) {
+    const path = url.split(matchedGerritUrl)[1];
+    const replaceRegex = path.startsWith('/gerrit/') ? /^\/gerrit\// : /^\//;
+    webUrl = `https://${matchedGerritUrl}/gerrit/gitweb?p=${path.replace(replaceRegex, '')};a=summary`;
+  } else {
+    webUrl = webUrl.replace(protocolRegex, 'https://');
```
Comment (patrikk0123) [inline src/utils/urlParseHelper.ts:45]: protocolRegex => protocol

**Review by patrikk0123 — COMMENTED** [review]
The merge of the two functions was a good choice. Also the code is now more readable.

File: src/utils/urlParseHelper.ts [hunk 2]
```diff
@@ -2,58 +2,48 @@ interface IScmRepositoryUrl {
...
-  if (['http', 'https', '@'].some((element) => url.includes(element))) {
-    const urlRes = url.includes('@') ? 'https://' + url.split('@').at(1)?.replace(':', '/') : url;
-    const base = urlRes.split('://').at(1)?.split('/').at(0) || '';
-    return { scmRepositoryUrl: url, webUrl: urlRes, name: base };
-  }
-  if (url.includes('.git')) {
-    const urlRes = url;
-    const base = urlRes.split('/').at(0) || '';
-    return { scmRepositoryUrl: url, webUrl: urlRes, name: base };
+  let webUrl = protocol === 'git@' ? url.replace(':', '/') : url;
```
[the remark this answers is not in the document]
Comment (DnsZhou) [reply src/utils/urlParseHelper.ts:34]: I don't think this format will introduce `:` symbol. see external SCM of /pnc-web/#/scm-repositories/514 from dev env as an example.

File: src/utils/urlParseHelper.ts [hunk 3]
```diff
@@ -2,58 +2,49 @@ interface IScmRepositoryUrl {
...
-    : url.split(base + (['https', 'http'].includes(protocol) ? '/gerrit/' : '/')).at(1);
-  const webUrl =
-    isGitProtocol || url.includes(protocol + '://' + gitlabBase)
-      ? 'https://' + base + '/' + project
-      : 'https://' + base + '/gerrit/gitweb?p=' + project + ';a=summary';
-  const name = url.includes(gitlabBase) ? 'GitLab' : 'Gerrit';
-  return { scmRepositoryUrl: url, webUrl, name };
-};
+export const parseScmRepositoryUrl = ({ url }: IScmRepositoryUrl): IParsedUrl => {
+  const protocolMatch = url.match(protocolRegex) || '';
```
Comment (patrikk0123) [inline src/utils/urlParseHelper.ts:32]: `match` returns array, therefore it should be `|| []`

**Review by patrikk0123 — APPROVED** [review]

**Pull request merged**